doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,553 @@
 Solveset
-==========
+========
 
 .. module:: sympy.solvers.solveset
 
-This module contains functions to solve a single equation for a single variable.
+This is the official documentation of the ``solveset`` module in solvers.
+It contains the frequently asked questions about our new module to solve
+equations.
+
+What's wrong with solve():
+--------------------------
+
+SymPy already has a pretty powerful ``solve`` function. But it has a lot of major
+issues
+
+1. It doesn't have a consistent output for various types of solutions
+   It needs to return a lot of types of solutions consistently:
+
+   * Single solution : `x = 1`
+   * Multiple solutions: `x^2 = 1`
+   * No Solution: `x^2 + 1 = 0 ; x \in \mathbb{R}`
+   * Interval of solution: `\lfloor x \rfloor = 0`
+   * Infinitely many solutions: `sin(x) = 0`
+   * Multivariate functions with point solutions: `x^2 + y^2 = 0`
+   * Multivariate functions with non-point solution: `x^2 + y^2 = 1`
+   * System of equations: `x + y = 1` and `x - y = 0`
+   * Relational: `x > 0`
+   * And the most important case "We don't Know"
+
+2. The input API is also a mess, there are a lot of parameters. Many of them
+   are not needed and they make it hard for the user and the developers to
+   work on solvers.
+
+3. There are cases like finding the maxima and minima of function using
+   critical points where it is important to know if it has returned all the
+   solutions. ``solve`` does not guarantee this.
+
+
+Why Solveset?
+-------------
+
+* ``solveset`` has a cleaner input and output interface: ``solveset`` returns
+  a set object and a set object takes care of all types of output. For
+  cases where it doesn't "know" all the solutions a ``ConditionSet`` with partial
+  solution is returned. For input it only takes the equation, the variables
+  to solve for and the optional argument ``domain`` in which the equations has to
+  be solved.
+
+* ``solveset`` can return infinitely many solutions. For example solving for
+  `\sin{(x)} = 0` returns `\{2 n \pi | n \in \mathbb{Z}\} \cup \{2 n \pi + \pi | n \in \mathbb{Z}\}`,
```

Comment:
I think that may work, but not sure how far it would go, we have to see that, since we are using `ImageSet` for manipulating infinite solutions. In `solveset` trigonometric and hyperbolic equations are solved by rewriting them into `exp` form and solving them using polys. `exp` is [inverted as follows](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L263):

```python
    if isinstance(f, exp):
        if isinstance(g_ys, FiniteSet):
            exp_invs = Union(*[imageset(Lambda(n, I*(2*n*pi + arg(g_y)) +
                                               log(Abs(g_y))), S.Integers)
                               for g_y in g_ys if g_y != 0])
```